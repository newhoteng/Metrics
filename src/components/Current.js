import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { BsArrowRightCircle, BsWind } from 'react-icons/bs';
import { IoIosSettings } from 'react-icons/io';
import { MdKeyboardVoice } from 'react-icons/md';
import { getCurrentAQIs } from '../redux/current/currentSlice';
// BsChevronLeft

import styles from '../styles/Current.module.css';

function Current() {
  const { currentAQIs, isLoading, error } = useSelector((store) => store.currentAQIs);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!currentAQIs.length) {
      dispatch(getCurrentAQIs());
    }
  }, [dispatch, currentAQIs]);

  if (isLoading) {
    return (
      <div style={{ padding: '3rem' }}>
        <h1>Loading...</h1>
      </div>
    );
  }
  if (error) {
    return (
      <div style={{ padding: '3rem' }}>
        <h1>Something went wrong</h1>
      </div>
    );
  }

  return (
    <>
      <header className={styles.header}>
        <div>current air quality</div>
        <div>
          <MdKeyboardVoice />
          <IoIosSettings />
        </div>
      </header>
      <main className={styles.main}>
        <div className={styles.hero}>
          <div />
          <div>
            <p>SELECTED CITIES</p>
            <p>air quality index</p>
          </div>
        </div>
        <div className={styles.banner}>INDEX BY CITY</div>
        <div className={styles.countriesContainer}>
          {currentAQIs.map((city) => (
            <Link to={`/${city.name}`} key={city.id} className={styles.countryLink}>
              <BsArrowRightCircle className={styles.arrowCircle} />
              <BsWind className={styles.airIcon} style={{ color: `${city.colorIndicator}` }} />
              <div className={styles.linkText}>
                <p>{city.name}</p>
                <p>
                  aqi:
                  {' '}
                  {city.aqi}
                  {' '}
                  -
                  {' '}
                  {city.textValue}
                </p>
              </div>
            </Link>
          ))}
        </div>
      </main>
    </>
  );
}

export default Current;
