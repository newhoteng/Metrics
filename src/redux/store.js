import { configureStore } from '@reduxjs/toolkit';
import currentAQIsReducer from './current/currentSlice';

export const store = configureStore({
  reducer: {
    currentAQIs: currentAQIsReducer,
  },
});

export default store;
