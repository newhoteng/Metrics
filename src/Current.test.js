import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from './redux/store';

// Component imports
import Current from './components/Current';

it('current aqi page renders correctly', () => {
  const currentAQIPage = renderer
    .create(
      <Provider store={store}>
        <Current page="/">Home Page</Current>
      </Provider>,
    )
    .toJSON();
  expect(currentAQIPage).toMatchSnapshot();
});

it('renders Current component', () => {
  render(
    <Provider store={store}>
      <Current />
    </Provider>,
  );
});