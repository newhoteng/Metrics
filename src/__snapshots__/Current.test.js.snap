// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`current aqi page renders correctly 1`] = `
Array [
  <header
    className="header"
  >
    <nav>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
          fillRule="evenodd"
        />
      </svg>
      <div>
        2023
      </div>
    </nav>
    <div>
      air quality
    </div>
    <div>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
        />
      </svg>
    </div>
  </header>,
  <main
    className="main"
  >
    <div
      className="hero"
    >
      <div />
      <div>
        <p>
          SELECTED CITIES
        </p>
        <p>
          air quality index
        </p>
      </div>
    </div>
    <div
      className="banner"
    >
      INDEX BY CITY
    </div>
    <div
      className="countriesContainer"
    />
  </main>,
]
`;
